jib-core/src/main/java/com/google/cloud/tools/jib/api/ImageReference.java

```diff
@@ -380,7 +411,7 @@ private String toString(boolean includeTagIfDigestPresent) {
 
     if (Strings.isNullOrEmpty(digest) || includeTagIfDigestPresent) {
```

Comment:
> toStringWithQualifier() is simply toString(false) + getQualifier(). No conditionally needed in the method.

I think there's still some conditionality needed here to add either a `:` or an `@` depending on whether it's a tag or digest.

Reply:
Ah, yeah, I missed the `:` and `@`. I still think we can make this simpler with

- `toString() { return toString(false); }`
- `toStringWithQualifier() { return toString(true); }`

`toString(boolean)` is already complex with conditionals and exists only because of `toStringWithQualifier()`. Moreover, I think the conditional connotation in the name `includeTagIfDigestPresent` makes it not easy to reason the implementation. I don't find a good reason to have a separate conditional in `toStringWithQualifier()`.

Reply:
But what I just said may not necessarily be true. Just try the basic idea and if it adversely makes it complex, just forgo the idea.